carbon-design-system/carbon
# fix(date-picker): correct deleted end range behavior

Closes #12734 

Prevents end date from being deleted on datepicker change when user has not made changes to input. 

#### Changelog

**Changed**

- packages/react/src/components/DatePicker/DatePicker.js: extend `onChange` behavior to prevent end date being deleted


#### Testing / Reviewing

1.  Tab to the Range Datepicker
2. Type a valid date into the "Start Date"
3. Press tab (focus now moves to the "End Date" and keeps the start date)
4. Type a valid date into the "End Date"
5.  Tab back to Start Date (shift + tab) --  you'll have to do it twice to get to the start date, see note below
6. Tab again to End Date (Tab)

You should still see both previously selected dates displayed in the calendar. Compare to react.carbondesignsystem.com where date gets deleted


### Note

There's a "double focus" issue when tabbing off the end date after changing it manually and not pressing "enter" to trigger the change, I paired up with Taylor on this and we're fine with this behavior for now until flatpickr gets more stable upstream or we switch into a different implementation
